cpp/examples/arrow/exec_plan_examples.cc

```diff
@@ -0,0 +1,1122 @@
+      gen = arrow::MakeTransferredGenerator(std::move(gen),
+                                            ::arrow::internal::GetCpuThreadPool());
+    } else {
+      gen = arrow::MakeVectorGenerator(std::move(opt_batches));
+    }
+
+    return gen;
+  }
+};
+
+BatchesWithSchema MakeBasicBatches() {
+  BatchesWithSchema out;
+  out.batches = {GetExecBatchFromJSON({arrow::int32(), arrow::boolean()},
+                                      "[[null, true], [4, false]]"),
+                 GetExecBatchFromJSON({arrow::int32(), arrow::boolean()},
+                                      "[[5, null], [6, false], [7, false]]")};
+  out.schema = arrow::schema(
+      {arrow::field("a", arrow::int32()), arrow::field("b", arrow::boolean())});
+  return out;
+}
+
+BatchesWithSchema MakeSortTestBasicBatches() {
+  BatchesWithSchema out;
+  out.batches = {
+      GetExecBatchFromJSON(
+          {arrow::int32(), arrow::int32(), arrow::int32(), arrow::int32()},
+          "[[1, 3, 0, 2], [121, 101, 120, 12], [10, 110, 210, 121], [51, 101, 2, 34]]"),
+      GetExecBatchFromJSON(
+          {arrow::int32(), arrow::int32(), arrow::int32(), arrow::int32()},
+          "[[11, 31, 1, 12], [12, 101, 120, 12], [0, 110, 210, 11], [51, 10, 2, 3]]")
+  };
+  out.schema = arrow::schema(
+      {arrow::field("a", arrow::int32()), arrow::field("b", arrow::int32()),
+       arrow::field("c", arrow::int32()), arrow::field("d", arrow::int32())});
+  return out;
+}
+
+BatchesWithSchema MakeGroupableBatches(int multiplicity = 1) {
+  BatchesWithSchema out;
+
+  out.batches = {GetExecBatchFromJSON({arrow::int32(), arrow::utf8()}, R"([
+                   [12, "alfa"],
+                   [7,  "beta"],
+                   [3,  "alfa"]
+                 ])"),
+                 GetExecBatchFromJSON({arrow::int32(), arrow::utf8()}, R"([
+                   [-2, "alfa"],
+                   [-1, "gama"],
+                   [3,  "alfa"]
+                 ])"),
+                 GetExecBatchFromJSON({arrow::int32(), arrow::utf8()}, R"([
+                   [5,  "gama"],
+                   [3,  "beta"],
+                   [-8, "alfa"]
+                 ])")};
+
+  size_t batch_count = out.batches.size();
+  for (int repeat = 1; repeat < multiplicity; ++repeat) {
+    for (size_t i = 0; i < batch_count; ++i) {
+      out.batches.push_back(out.batches[i]);
+    }
+  }
+
+  out.schema = arrow::schema(
+      {arrow::field("i32", arrow::int32()), arrow::field("str", arrow::utf8())});
+
+  return out;
+}
+
+std::shared_ptr<arrow::internal::ThreadPool> MakeIOThreadPool() {
+  auto maybe_pool = arrow::internal::ThreadPool::MakeEternal(/*threads=*/8);
+  if (!maybe_pool.ok()) {
+    maybe_pool.status().Abort("Failed to create global IO thread pool");
+  }
+  return *std::move(maybe_pool);
+}
+
+arrow::internal::ThreadPool* GetIOThreadPool() {
+  static std::shared_ptr<arrow::internal::ThreadPool> pool = MakeIOThreadPool();
+  return pool.get();
+}
```

Comment:
This seems unused.

Reply:
Need to clean up some of these functions

Reply:
Removed.